[CODEMOD][caffe2] use npt.NDArray instead of np.ndarray in type annotations

igorsugak opened the pull request:
Summary:
To facilitate PSS-2 upgrade, this uses `ndt.NDArray` instead of `nd.ndarray` in type annotations. In Numpy-1.19 (PSS-1) it's an alias to `nd.ndarray` -- a noop.
In Numpy-1.24, `ndt.NDArray` a proper generic type, and without this change uses of `nd.ndarray` generate this Pyre type error:
```counterexample
 Invalid type parameters [24]: Generic type `np.ndarray` expects 2 type parameters.
```

Test Plan: Sandcastle plus visual inspection

Differential Revision: D62977370


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

justinchuby commented on torch/onnx/_internal/exporter/_core.py:
```diff
@@ -43,6 +43,7 @@
     import os
 
     import numpy as np
+import numpy.typing as npt
```
```suggestion
    import numpy.typing as npt
```

igorsugak committed:
[CODEMOD][caffe2] use npt.NDArray instead of np.ndarray in type annotations (#136288)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/136288

To facilitate PSS-2 upgrade, this uses `ndt.NDArray` instead of `nd.ndarray` in type annotations. In Numpy-1.19 (PSS-1) it's an alias to `nd.ndarray` -- a noop.
In Numpy-1.24, `ndt.NDArray` a proper generic type, and without this change uses of `nd.ndarray` generate this Pyre type error:
```counterexample
 Invalid type parameters [24]: Generic type `np.ndarray` expects 2 type parameters.
```

Test Plan: Sandcastle plus visual inspection

Reviewed By: kit1980

Differential Revision: D62977370

jeanschmidt commented:
@pytorchbot merge -f "landed internally"
